library/core/src/primitive_docs.rs

```diff
@@ -1211,6 +1216,72 @@ mod prim_f16 {}
 /// *[See also the `std::f32::consts` module](crate::f32::consts).*
 ///
 /// [wikipedia]: https://en.wikipedia.org/wiki/Single-precision_floating-point_format
+///
+/// # NaN bit patterns
+///
+/// This section defines the possible NaN bit patterns returned by non-"bitwise" floating point
+/// operations. The bitwise operations are unary `-`, `abs`, `copysign`; those are guaranteed to
+/// exactly preserve the bit pattern of their input except for possibly changing the sign bit.
+///
+/// A floating-point NaN value consists of:
+/// - a sign bit
+/// - a quiet/signaling bit
+/// - a payload, which makes up the rest of the significand (i.e., the mantissa) except for the
+///   quiet/signaling bit.
```

Comment:
Maybe this should mention the exponent part too?

Reply:
What should it say there? The point of this section is to document the degrees of freedom of a NaN value, and the exponent is fixed.

Reply:
Right, but the "consists of" list isn't complete if it doesn't specify how the NaN is tagged as a NaN, which happens in the exponent bits (combined with at least one nonzero mantissa bit). Otherwise it's like describing Rust enums without mentioning the discriminator.

Reply:
A Some value "consists of" the `T` that it carries in its field. Similarly, a NaN consists of the ingredients listed here. The discriminator is what decides whether an "Option" value is a "Some" value, and whether a float value is a NaN value, but it is not part of the Some/Option value.